### PR TITLE
[RFC] api.c: add support for empty cgroup v2 deletion

### PR DESCRIPTION
**Pull request by kamalesh-babulal**
Add support to recongize empty cgroup v2 with no controllers attached to
it in cgroup_delete_cgroup_ext(), that gets called to remove a cgroup.

In the deletion path, cgroup_build_tasks_procs_path() builds the path
of the cgroup for re-assignment/movement of tasks from the cgroup that's
getting deleted, teach it to recongize the empty controller as cgroup
V2.

Fixes: https://github.com/libcgroup/libcgroup/issues/125

Signed-off-by: Kamalesh Babulal <kamalesh.babulal@oracle.com>

### CODE REVIEW

**Comment by kamalesh-babulal** [comment]
**V2 Changes**
- Added check before `controller_name` assignment.

File: src/api.c [hunk 1]
```diff
@@ -3068,19 +3070,31 @@ int cgroup_delete_cgroup_ext(struct cgroup *cgroup, int flags)
...
 	}
 
-	/* Remove the group from all controllers. */
-	for (i = 0; i < cgroup->index; i++) {
+	/*
+	 * Remove the group from all controllers and in the case of cgroup
+	 * with no controllers, perform all actions of a single controller.
+	 */
+	for (i = 0;
+	     empty_cgroup > 0 || i < cgroup->index; i++, empty_cgroup--) {
```
Comment (drakenclimber) [inline src/api.c:3087]: nifty :)

**Review by drakenclimber — APPROVED** [review]

**Comment by drakenclimber** [comment]
Pushed to main in commit ca55c929e8074e630f3fd2eded96542b88ef1ad5.  Thanks, @kamalesh-babulal

File: src/api.c (hunk 1 above)
Comment (kamalesh-babulal) [reply src/api.c:3087]: Thanks for the review @drakenclimber

**Pull request closed without merging**